### PR TITLE
V7.1 update

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
added:
- add task debug doc
- add app developer secret doc
- add requester secret doc

changed:
- update required versions
- change target chain from viviani to bellecour

cleanup:
- remove some unused pages
- format MD

### CODE REVIEW

**Comment by PierreJeanjacquot** [comment]
live preview: https://documentations.iex.ec/staging/v/v7.1-update/

File: for-developers/confidential-computing/app-developer-secret.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+**Prerequisites**
+
+* [Docker](https://docs.docker.com/install/) 17.05 or higher on the daemon and client.
+* [Nodejs](https://nodejs.org) 14.0.0 or higher.
+* [iExec SDK](https://www.npmjs.com/package/iexec) 7.2.0 or higher.
+* Familiarity with the basic concepts of [Intel® SGX](intel-sgx-technology.md#intel-r-software-guard-extension-intel-r-sgx) and [SCONE](intel-sgx-technology.md#scone-framework) framework.
+{% endhint %}
+
+{% hint style="warning" %}
+Please make sure you have already checked the [Quickstart](../your-first-app.md), [Your first application](../your-first-app.md) and [Build trusted applications](create-your-first-sgx-app.md) tutorials before learning how to use a developer secret.
```
Comment (jbern0rd) [inline for-developers/confidential-computing/app-developer-secret.md]: Wrong file for Quickstart link ? Shouldn't it be quick-start-for-developers.md ?

File: for-developers/confidential-computing/app-developer-secret.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+{% endhint %}
+
+Trusted Execution Environments offer a huge advantage from a security perspective. They guarantee that the behavior of execution does not change even when launched on an untrusted remote machine. The data inside this type of environment is also protected, which allows its monetization while preventing leakage.
+
+With iExec, it is possible to securely to associate an application developer secret to the runtime of an application.
```
Comment (jbern0rd) [inline for-developers/confidential-computing/app-developer-secret.md]:
```suggestion
With iExec, it is possible to securely associate an application developer secret to the runtime of an application.
```

File: for-developers/confidential-computing/app-developer-secret.md [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+
+{% hint style="warning" %}
+The app developer secret is only exposed to your app inside authorized [enclaves](intel-sgx-technology.md#enclave) and never leave them.
```
Comment (jbern0rd) [inline for-developers/confidential-computing/app-developer-secret.md]:
```suggestion
The app developer secret is only exposed to your app inside authorized [enclaves](intel-sgx-technology.md#enclave) and never leaves them.
```

File: for-developers/confidential-computing/app-developer-secret.md [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+{% endhint %}
+
+Let's see how to do all of that!
+
+## Prepare your application
+
+{% hint style="info" %}
+We will use the API [countapi.xyz](https://countapi.xyz/).
+This service keeps a count of hit on any couple of `namespace/key` (ex: <https://api.countapi.xyz/hit/foo/bar>).
+In this example, we will use an app developer secrets to set `namespace/key`.
```
Comment (jbern0rd) [inline for-developers/confidential-computing/app-developer-secret.md]:
```suggestion
This service keeps a count of hits on any couple of `namespace/key` (ex: <https://api.countapi.xyz/hit/foo/bar>).
In this example, we will use an app developer secret to set `namespace/key`.
```

File: for-developers/confidential-computing/app-developer-secret.md [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+mkdir src
+touch Dockerfile
+touch sconify.sh
+```
+
+{% endcode %}
+{% endtab %}
+{% endtabs %}
+
+The application use the developer secret to make a call to a secret endpoint of [countapi.xyz](https://countapi.xyz/) and writes the result in a file:
```
Comment (jbern0rd) [inline for-developers/confidential-computing/app-developer-secret.md]:
```suggestion
The application uses the developer secret to make a call to a secret endpoint of [countapi.xyz](https://countapi.xyz/) and writes the result in a file:
```

File: for-developers/confidential-computing/requester-secrets.md [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+**Prerequisites**
+
+* [Docker](https://docs.docker.com/install/) 17.05 or higher on the daemon and client.
+* [Nodejs](https://nodejs.org) 14.0.0 or higher.
+* [iExec SDK](https://www.npmjs.com/package/iexec) 7.2.0 or higher.
+* Familiarity with the basic concepts of [Intel® SGX](intel-sgx-technology.md#intel-r-software-guard-extension-intel-r-sgx) and [SCONE](intel-sgx-technology.md#scone-framework) framework.
+{% endhint %}
+
+{% hint style="warning" %}
+Please make sure you have already checked the [Quickstart](../your-first-app.md), [Your first application](../your-first-app.md) and [Build trusted applications](create-your-first-sgx-app.md) tutorials before learning how to use requester secrets.
```
Comment (jbern0rd) [inline for-developers/confidential-computing/requester-secrets.md]: Same question for Quickstart link. Shouldn't it be `../quick-start-for-developers.md` ?

File: for-developers/confidential-computing/requester-secrets.md [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+{% endhint %}
+
+Trusted Execution Environments offer a huge advantage from a security perspective. They guarantee that the behavior of execution does not change even when launched on an untrusted remote machine. The data inside this type of environment is also protected, which allows its monetization while preventing leakage.
+
+With iExec, it is possible to securely to associate an application developer secret to the runtime of an application.
```
Comment (jbern0rd) [inline for-developers/confidential-computing/requester-secrets.md]: Rephrase this sentence for requesters ?

File: for-developers/confidential-computing/requester-secrets.md [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+
+Let's create a directory tree for this app in `~/iexec-projects/`.
+
+{% tabs %}
+{% tab title="Javascript" %}
+{% code %}
+
+```bash
+cd ~/iexec-projects
+mkdir my-tee-requester-secrets-app && cd my-tee-requester-secrets-app
```
Comment (jbern0rd) [inline for-developers/confidential-computing/requester-secrets.md]:
```suggestion
mkdir tee-requester-secrets-app && cd tee-requester-secrets-app
```
for naming convention consistency ?

File: for-developers/confidential-computing/requester-secrets.md [hunk 9]
```diff
@@ -0,0 +1,407 @@
...
+```
+
+* the secret named `foo` will be available in `IEXEC_REQUESTER_SECRET_1`
+* the secret named `bar` will be available in `IEXEC_REQUESTER_SECRET_3`
+* `IEXEC_REQUESTER_SECRET_2` will be skipped
+{% endhint %}
+
+## Next step?
+
+Thanks to the explained confidential computing workflow, it is possible to use consume requester secrets with a trusted application.
```
Comment (jbern0rd) [inline for-developers/confidential-computing/requester-secrets.md]: Fix `to use consume requester secrets`

File: for-developers/your-first-app.md [hunk 10]
```diff
@@ -47,89 +46,207 @@ Today you can run any application as a task. This means services are not support
...
+
+##### Provisioning an app developer secret
+
+The app developer pushes an **app developer secret** to the Secret Management Service.
+
+Once pushed, an **app developer secret** cannot be modified.
+
+##### Consuming an app developer secret
+
+The **app developer secret** is exposed to the application in `IEXEC_APP_SECRET`
```
Comment (jbern0rd) [inline for-developers/your-first-app.md]:
```suggestion
The **app developer secret** is exposed to the application in `IEXEC_APP_DEVELOPER_SECRET`
```

File: for-developers/your-first-app.md [hunk 11]
```diff
@@ -47,89 +46,207 @@ Today you can run any application as a task. This means services are not support
...
+  ├── stdout.txt
+  ├── stderr.txt
+  ├── computed.json
+  └── ...
 ```
 
-`stdout.txt` contains the logs of your application. This file is automatically generated by the iExec worker.
-
-`iexec_out` is a copy at the final state of the container's iExec output directory.
+* The iExec worker automaticaly creates `stdout.txt` and `stdout.txt` containing the logs of your application.
```
Comment (jbern0rd) [inline for-developers/your-first-app.md]: `stdout.txt` and `stderr.txt` ?

File: for-developers/confidential-computing/app-developer-secret.md (hunk 1 above)
Comment (PierreJeanjacquot) [reply for-developers/confidential-computing/app-developer-secret.md]: good catch, fixed 9b75cbd2945d5924690cfb021e6fee707760834b

File: for-developers/your-first-app.md (hunk 11 above)
Comment (PierreJeanjacquot) [reply for-developers/your-first-app.md]: yes 80b50cfa75ef4166d2ddb135bd98d0fb5cba5af7

File: for-developers/confidential-computing/requester-secrets.md (hunk 9 above)
Comment (PierreJeanjacquot) [reply for-developers/confidential-computing/requester-secrets.md]: fixed a00bac52dc76c2b15044a7796daf5dbdea849d51

File: for-developers/confidential-computing/requester-secrets.md (hunk 7 above)
Comment (PierreJeanjacquot) [reply for-developers/confidential-computing/requester-secrets.md]: missed this one, thank you @jbern0rd 
what do you think of this?
```suggestion
With iExec, it is possible to securely consume requester-provided secrets in the application.
```

**Comment by izimio** [comment]
# Javascript label for a code area wrongly used : 
## Juste below the title in [Confidential Computing/Use a developper secret](https://documentations.iex.ec/staging/v/v7.1-update/for-developers/confidential-computing/app-developer-secret#prepare-your-application)

## Same mistake one page further at : [Confidential Computing/Use a developper secret](https://documentations.iex.ec/staging/v/v7.1-update/for-developers/confidential-computing/requester-secrets#prepare-your-application)

## Favor using ```bash``` or no label at all,  juste like in the [build trusted application](https://documentations.iex.ec/staging/v/v7.1-update/for-developers/confidential-computing/create-your-first-sgx-app#prepare-your-application) page 

image

**Comment by PierreJeanjacquot** [comment]
thank you @izimio fixed e36b3710b11e150d713aab6e8f50575c030be98a

File: for-developers/confidential-computing/sgx-encrypted-dataset.md [hunk 12]
```diff
@@ -132,55 +132,36 @@ For simplicity, we will use the dataset with a TEE-debug app on a debug workerpo
...
+## Prepare your application
 
 Let's create a directory tree for this app in `~/iexec-projects/`.
 
-{% tabs %}
-{% tab title="Javascript" %}
-{% code %}
 ```bash
 cd ~/iexec-projects
 mkdir my-tee-dataset-app && cd my-tee-dataset-app
```
Comment (jbern0rd) [inline for-developers/confidential-computing/sgx-encrypted-dataset.md]: For consistency with sconify.sh
```suggestion
mkdir tee-dataset-app && cd tee-dataset-app
```

**Pull request merged**